daemon/daemon.go

```diff
@@ -1142,6 +1142,14 @@ func (daemon *Daemon) GetRemappedUIDGID() (int, int) {
 // created. nil is returned if a child cannot be found. An error is
 // returned if the parent image cannot be found.
 func (daemon *Daemon) ImageGetCached(imgID string, config *runconfig.Config) (*image.Image, error) {
+	// for now just exit if imgID has no children.
+	// maybe parentRefs in graph could be used to store
+	// the Image obj children for faster lookup below but this can
+	// be quite memory hungry.
+	if !daemon.Graph().HasChildren(imgID) {
+		return nil, nil
+	}
+
```

Comment:
`HasChildren` calls `ByParent` internally. `ByParent` returns a map indexed by the parent ID, which is similar to what this function constructs below. Couldn't this function call `ByParent`, check `len(parents[imgID])`, and if it's nonzero, use the map returned by `ByParent` to find the earliest created child? That would avoid an extra walk of the graph.